app/src/test/java/org/simple/clinic/home/patients/PatientsScreenControllerTest.kt

```diff
@@ -108,43 +107,26 @@ class PatientsScreenControllerTest {
   }
 
   @Test
-  fun `when screen is created and the user is awaiting approval then the approval status should be checked`() {
-    val user = PatientMocker.loggedInUser(status = WaitingForApproval)
-    whenever(userSession.loggedInUser()).thenReturn(Observable.just(Just(user)))
-    whenever(userSession.refreshLoggedInUser()).thenReturn(Completable.complete())
-    whenever(userSession.canSyncData()).thenReturn(Observable.never())
-    whenever(hasUserDismissedApprovedStatus.asObservable()).thenReturn(Observable.just(false))
-    whenever(hasUserDismissedApprovedStatus.get()).thenReturn(false)
-
-    uiEvents.onNext(ScreenCreated())
-    uiEvents.onNext(TheActivityLifecycle.Resumed())
-    uiEvents.onNext(TheActivityLifecycle.Resumed())
-
-    verify(userSession, times(3)).refreshLoggedInUser()
-  }
-
-  @Test
-  @Parameters(method = "params for approved and disapproved for syncing")
-  fun `when screen is created and the user is not awaiting approval then the user's status should not be checked`(
+  @Parameters(method = "params for user status for syncing")
+  fun `when screen is created or resumed then the user's status should refresh regardless of current status`(
       status: UserStatus
   ) {
     val user = PatientMocker.loggedInUser(status = status)
     whenever(userSession.loggedInUser()).thenReturn(Observable.just(Just(user)))
-    whenever(userSession.refreshLoggedInUser()).thenReturn(Completable.never())
+    whenever(userSession.refreshLoggedInUser()).thenReturn(Completable.complete())
     whenever(approvalStatusApprovedAt.get()).thenReturn(Instant.now())
     whenever(hasUserDismissedApprovedStatus.asObservable()).thenReturn(Observable.just(false))
     whenever(hasUserDismissedApprovedStatus.get()).thenReturn(false)
 
     uiEvents.onNext(ScreenCreated())
     uiEvents.onNext(TheActivityLifecycle.Resumed())
-    uiEvents.onNext(TheActivityLifecycle.Resumed())
 
-    verify(userSession, never()).refreshLoggedInUser()
+    verify(userSession, times(2)).refreshLoggedInUser()
```

Comment:
It's not obvious from this check that each of the two different `UiEvent`s (`ScreenCreated` and `Resumed`) can independently trigger a user refresh. Can we either:
1. Pass the event in via the test parameter as well so we can test each event separately, or
2. Make one verification call after the first event, and another after the second event so we know that each event triggers a refresh independently.

```kotlin
uiEvents.onNext(ScreenCreated())
verify(userSession).refreshLoggedInUser()

clearInvocations(userSession)

uiEvents.onNext(TheActivityLifecycle.Resumed())
verify(userSession).refreshLoggedInUser()
```

Reply:
Good point.